syncstorage/tests/functional/support.py

```diff
@@ -50,7 +51,14 @@ def _authenticate(self):
         self.user_id = random.randint(1, 100000)
         auth_policy = self.config.registry.getUtility(IAuthenticationPolicy)
         req = Request.blank(self.host_url)
-        creds = auth_policy.encode_hawk_id(req, self.user_id)
+        creds = auth_policy.encode_hawk_id(
+            req, self.user_id, extra={
+                # Include a hashed_fxa_uid to trigger uid/kid extraction
+                "hashed_fxa_uid": str(uuid.uuid4()),
```

Comment:
I just noticed this annoyance as well while working on the loadtest update, I wonder if there's any better way to detect that "fxa_uid" is the new-style thing rather than the old hashed thing.  Maybe we could switch on the presence of an `fxa_kid` key instead, over in https://github.com/mozilla-services/server-syncstorage/blob/master/syncstorage/views/authentication.py#L182 ?

Reply:
So, all the validation logic in authentication.py is a bit of a mess. If you're ok with introducing a new dependency, I'd be happy to allow optional extraction/validation of more of these attributes without repeating the validation code over and over with Marshmallow (https://github.com/marshmallow-code/marshmallow).

Reply:
Sounds good to me!

Reply:
I'm going to bring the marshmallow validation in with a separate PR/commit as it'll likely shuffle a lot of stuff around unrelated to Spanner.